### PR TITLE
Clean up English in JSON documentation

### PR DESCRIPTION
**Pull request by AdamLearns**
This change only improves the wording and grammar used on [this page](https://docs.godotengine.org/en/stable/classes/class_json.html). I originally only wanted to fix a typo, but I ended up proof-reading the whole page, and at least one function should be clearer as a result.

Note that I did not update any translation files:

```
╰─❯ rg "enables all data"
doc/translations/zh_TW.po
50968:"The [JSON] enables all data types to be converted to and from a JSON string. "

doc/translations/zh_CN.po
68886:"The [JSON] enables all data types to be converted to and from a JSON string. "
```

...should something be done with those?

### CODE REVIEW

**Comment by Calinou** [comment]
> Note that I did not update any translation files:
> ...should something be done with those?

You don't need to do anything here, since they are updated automatically by maintainers on a regular basis.

File: doc/classes/JSON.xml
Comment (Mickeon) [inline doc/classes/JSON.xml:56]: You cannot reference parameters from other functions this way, at most you'd have to use `[code]` tags

Comment (AThousandShips) [reply doc/classes/JSON.xml:56]:
```suggestion
				Return the text parsed by [method parse] (requires [code]keep_text[/code] to be passed to [method parse]).
```
Or similar

**Comment by AdamLearns** [comment]
Is there an easy way to see the resulting page that I've modified before committing? It didn't seem to be part of building Godot, and I wasn't positive based on the instructions [here](https://docs.godotengine.org/en/stable/contributing/documentation/contributing_to_the_documentation.html) if I just need to look at individual pages using Sphinx/reST.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks for contributing :) The commits need to be squashed into one before this can be merged. See [here](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase) for one explanation on how to do do that.

Otherwise, the changes themselves look good to me.

**Comment by akien-mga** [comment]
Thanks!

**Comment by AdamLearns** [comment]
Ah, sorry, Tuesdays and Thursdays are my catch-up days, so I didn't have a chance to look at this until now. I see that it's already been merged though—thank you for including the work! 🙏

**Pull request merged**
